linera-core/src/remote_node.rs

```diff
@@ -291,7 +291,19 @@ impl<N: ValidatorNode> RemoteNode<N> {
         if hashes.is_empty() {
             return Ok(Vec::new());
         }
-        self.node.download_certificates(hashes).await
+        let certificates = self.node.download_certificates(hashes.clone()).await?;
+        let returned = certificates
+            .iter()
+            .map(ConfirmedBlockCertificate::hash)
+            .collect();
+        ensure!(
+            returned == hashes,
```

Comment:
This expects the order to be the same as well, right? We probably maintain it but if it's different we'd fail here even if we actually got the same certs (just in different order).

Reply:
Yes. We do maintain the order, and I think it needs to be enforced. Callers of this method should be able to rely on the fact that they can e.g. `zip` their list of hashes and the certificates they got.